### ISSUE TITLE
[BUG] Video-Annotation Desync

### ISSUE THREAD

**Issue by BryonLewis** [Kitware/dive#421]
**Describe the bug**
Video is not in sync with the annotation data

**To Reproduce**
Public Data/ Folder: https://viame.kitware.com/#/viewer/5f5288d1683ae84c2a4f5612
Note that the annotations are not properly in sync.

**Related data**
I think we may need some concussion about the annotation/frame mismatch.  This seems complicated enough that a single solution wouldn't work.
3 Things going on with the video mismatching of annotations:

1.  Looks like we don't actually specify a frame rate based on the ffmpeg results after the conversion process - https://github.com/VIAME/VIAME-Web/blob/master/server/viame_tasks/tasks.py#L335-L340
2. Video returns as having a ~60fps frame rate from ffmpeg but it defaults to 5 because of the current code from above
3. We need a way to determine the frame rate of the source annotations for videos.  These annotations return frames, but when set to 60fps to match the video they didn't line up, but when set to 30 fps they were at least closer to the proper value.

**Expected behavior**
We want uploaded data to work with videos properly.  It is more difficult because there could be different Framerates between the video and the source data.

**Comment by subdavis** [comment]
The FPS of the actual video doesn't matter.  The FPS in the metadata only corresponds to the FPS at which the annotations were recorded.  For videos, as long as the FPS in the metadata represents the correct annotation FPS, everything will be fine.

Sync happens by taking the length of the video (in seconds) of the raw media file and pretending that its framerate is the annotation framerate.  So at t=10s, the annotations from frame 50 are shown.  Whether the video is 30 or 29.99 or 60fps, doesn't matter.  I think adding controls to let users set their framerate on videos would be good to have, but we should hide it in a menu and have some text that explains this.

TL;DR I don't think this is a bug.  Your open PR to fix image cache will also fix this.

**Comment by BryonLewis** [comment]
Thanks, I should of looked closer into the VideoAnnotator, I had in my mind how it worked without looking so that was my mistake.  I'm going to do some more checking and confirming tomorrow morning.

**Comment by BryonLewis** [comment]
It looks like annotations were edited at a different rate than they are being displayed.  They aren't perfect when adjusted and the invertebrate/vertebrate annotations are definitely meant to be at 5fps.  I don't know exactly how this happened, unless data was manually edited in another tool or added a different fps.  I have some references below to better illustrate.

5 fps:
![5fpsFishDatasmall](https://user-images.githubusercontent.com/61746913/98115668-2c68ba80-1e75-11eb-99f0-cbbc669b3168.gif)

30 fps:
![30fpsFishDatasmall](https://user-images.githubusercontent.com/61746913/98115684-34c0f580-1e75-11eb-8b19-fb66e8ab554a.gif)

**Comment by subdavis** [comment]
I assume the vertebrate/invertebrate came from our detector, and the rest came from another tool?  They imported a much of annotations @ 30hz then ran a pipeline?

What even happens when you do that?  You can't have annotations at different framerates.  I dont know what to do about this.

**Comment by BryonLewis** [comment]
Okay I think I have it now by going through his auxiliary folder and the commit history of VIAME-Web.  Before the update to setting the default frame rate to 10 it was previously set to 30 if something happened with the return value. (https://github.com/VIAME/VIAME-Web/blob/8a3635560e313facfe09603afcfae99f529a0a78/client/app/use/useGirderDataset.ts#L12-L42)  We probably didn't notice much in using image sequences because those are direct frames.  In videos if there is an issue you were viewing at 30 fps until a recent change which swapped it to defaulting to 10fps which is still off from the real value of 5fps.  So I believe that is how these discrepancies occurred.  I feel better now understanding what happened.

**Comment by subdavis** [comment]
How do we verify that fixing this bug hasn't broken people's annotations in the same way?

Right now, FPS is still being ignored, so all video annotations are being recorded at 10FPS.  Unless we go through an look at every annotated video, we've probably messed up a lot of people's data right?

**Comment by BryonLewis** [comment]
I don't know if we can just run a query for FPS value other than 10 because that wasn't the real issue, the real problem is that when they made those edits they did on an annotator that was at 30 or 10 fps where it didn't really take into account the fps of the metadata.
We may need to manually adjust video metadata to either 10 or 30 fps.  The case of running pipelines and then editing manually afterwards (even saving) could cause issues that aren't fixed by a simple fps change because the tracks are at different framerates.

**Comment by subdavis** [comment]
* I'm assuming there aren't other videos annotated at 30.
* But there might be videos that got annotated at 10 because of the bug.
* So once we fix the bug, videos annotated with 10 that have their true FPS set to 10 are fine.
* But once we fix the bug, videos annotated with 10 (by accident due to the bug) that have their true FPS set to something other than 10 will suddenly screw up.

That was my logic, at least.   

> We may need to manually adjust video metadata to either 10 or 30 fps. 

I figured that a query to see everything where FPS  !== 10 would give us all the candidates that might be wrong, that we could check manually and adjust manually if necessary.

>  The case of running pipelines and then editing manually afterwards (even saving) could cause issues that aren't fixed by a simple fps change because the tracks are at different framerates.

I think we need @mattdawkins here.  Users probably don't want their annotations converted to 5FPS just because they ran a pipeline, but we don't support FPS input to viame, so maybe we just have to lock all video at 5 for right now and not allow changing it? I don't know what the right answer here is.  It could be as simple as "Our public server supports pipelines at 5hz.  You want higher-frequency pipelines, you gotta run this on your own hardware".

**Comment by BryonLewis** [comment]
> I'm assuming there aren't other videos annotated at 30.

This dataset did this because the initial group of editing was done in late August (30fps) then updated around the beginning of October (10fps).  There could be others that have done the same but that is probably unlikely.  I think your logic is sound.

> Users probably don't want their annotations converted to 5FPS just because they ran a pipeline, but we don't support FPS input to viame, so maybe we just have to lock all video at 5 for right now and not allow changing it?

That was where I was heading as well after some thought.

**Comment by subdavis** [comment]
Very stale
